tests/integration/cloudformation/resources/test_apigateway.py

```diff
@@ -438,3 +442,111 @@ def test_api_gateway_with_policy_as_dict(deploy_cfn_template, snapshot, aws_clie
     rest_api["policy"] = json.loads(policy)
 
     snapshot.match("rest-api", rest_api)
+
+
+@pytest.mark.aws_validated
+@pytest.mark.skip_snapshot_verify(
+    paths=[
+        "$.put-ssm-param.Tier",
+        "$.get-resources.items..resourceMethods.GET",
+        "$.get-resources.items..resourceMethods.OPTIONS",
+        "$..methodIntegration.cacheNamespace",
+        "$.get-authorizers.items..authorizerResultTtlInSeconds",
+    ]
+)
+def test_rest_api_serverless_ref_resolving(
+    deploy_cfn_template, snapshot, aws_client, create_parameter, create_lambda_function
+):
+    template = """
+    AWSTemplateFormatVersion: '2010-09-09'
+    Transform: AWS::Serverless-2016-10-31
+    Description: AWS SAM template with a simple API definition
+    Parameters:
+      AllowedOrigin:
+        Type: 'String'
+      TestSSMParameter:
+        Type: 'AWS::SSM::Parameter::Value<String>'
+        Default: '/test-stack/testssm/random-value'
+      TestSSMParameterLambdaArn:
+        Type: 'AWS::SSM::Parameter::Value<String>'
+        Default: '/test-stack/testssm/lambda-arn'
+
+    Resources:
+      ApiGatewayApi:
+        Type: AWS::Serverless::Api
+        Properties:
+          StageName: local
+          OpenApiVersion: '2.0'
+          Cors:
+            AllowMethods: "'OPTIONS,POST,GET,PUT'"
+            AllowHeaders: !Sub "'Content-Type,Authorization,${TestSSMParameter}'"
+            AllowCredentials: true
+            AllowOrigin: !Sub "'${AllowedOrigin}'"
+          Auth:
+            Authorizers:
+              LambdaTokenAuthorizer:
+                FunctionArn: !Ref TestSSMParameterLambdaArn
+
+      ApiFunction: # Adds a GET api endpoint at "/" to the ApiGatewayApi via an Api event
+        Type: AWS::Serverless::Function
+        Properties:
+          Events:
+            ApiEvent:
+              Type: Api
+              Properties:
+                Path: /
+                Method: get
+                RestApiId:
+                  Ref: ApiGatewayApi
+          Runtime: python3.7
```

Comment:
nit: Would keep this at the latest version so we have more time until we need to update it :grin:  (if there's not another specific reason to use 3.7 here)

Reply:
Nope, just copied/pasted the sample from AWS docs. Changed 👍 thanks!